src/transformStateWithClones.js

```diff
@@ -1,13 +1,33 @@
-'use strict';
-
-/**
- * @param {Object} state
- * @param {Object[]} actions
- *
- * @return {Object[]}
- */
+"use strict";
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultArray = [{ ...state }];
+
+  for (let i = 0; i < actions.length; i++) {
+    const valueBeforePassing =
+      i === 0 ? { ...resultArray[i] } : { ...resultArray[i - 1] };
```

Comment:
You are creating a new copy of the state at the start of each iteration. This is unnecessary because you already have a copy from the previous iteration.